frontend/src/Metamaps/Listeners.js

```diff
@@ -122,7 +123,27 @@ const Listeners = {
     })
 
     $(window).resize(function () {
-      if (Visualize && Visualize.mGraph) Visualize.mGraph.canvas.resize($(window).width(), $(window).height())
+      if (Visualize && Visualize.mGraph){
+        //Find the current canvas scale and map-coordinate at the centre of the user's screen
+        var canvas = Visualize.mGraph.canvas,
```

Comment:
All further references to `Visualize.mGraph.canvas` after this variable declaration should just be simplified to `canvas`